File: python/rest/requests/get_flights_enroute.py
#!//usr/bin/env python

import requests

username = "YOUR_USERNAME"
apiKey = "YOUR_API_KEY"
fxmlUrl = "https://flightxml.flightaware.com/json/FlightXML3/"

payload = {'airport_code':'KSFO', 'type':'enroute', 'howMany':'10'}
response = requests.get(fxmlUrl + "AirportBoards", params=payload, auth=(username, apiKey))

if response.status_code == 200:
	print "Aircraft enroute to KSFO:"
	decodedResponse = response.json()
	for flight in decodedResponse['AirportBoardsResult']['enroute']['flights']:
		print "{} ({})\t{} ({})".format(flight['ident'], flight['aircrafttype'], flight['origin']['airport_name'], flight['origin']['code'])
else:
	print "There was an error retrieving the data from the server."

File: ruby/rest/rest-client/get_flights_enroute.rb
#!/usr/bin/env ruby

require 'rubygems'
require 'json'
require 'rest-client'
require 'base64'


username = 'YOUR_USERNAME'
apiKey = 'YOUR_API_KEY'

authHeader = Base64.strict_encode64("#{username}:#{apiKey}")
fxmlUrl = 'https://flightxml.flightaware.com/json/FlightXML3'

params = {
	:airport_code => 'KLAX',
	:include_ex_data => 0,
	:type => 'enroute',
	:howMany => 10
}

response = RestClient.get "#{fxmlUrl}/AirportBoards", {params: params, Authorization: "Basic #{authHeader}"}

if response.code == 200
	parsed = JSON.parse(response.body)

	puts "Flights enroute to KLAX: "
	flights = parsed["AirportBoardsResult"]["enroute"]["flights"]
	flights.each do |flight|
		puts "#{flight['ident']} (#{flight['aircrafttype']})\t#{flight['origin']['airport_name']} (#{flight['origin']['code']})"
	end
else
	puts "Error retrieving data from the server"
end